sass/static/js/sass_dashboard.js

```diff
@@ -890,9 +890,30 @@ $(function () {
     }
     $('.download-as-csv').click(onDownloadCSVClicked);
     $('.download-summary-as-csv').click(onDownloadSummaryCSVClicked);
-    $('.download-latest-as-csv').on('click', function () {
-        var filename = 'SASS_Taxa_per_biotope_' + sassLatestData;
-        exportTableToCSV(filename + '.csv', "sass-taxon-per-biotope-table")
+    $('.download-latest-as-csv').on('click', function (e) {
+        let downloadButton = $(e.target);
+        let includeAllTaxon = $('#include-all-taxon').prop('checked');
+        if (includeAllTaxon) {
+            let csv_name = getCsvName('All data', 'Sass taxon data');
+            let currentUrl = window.location.href;
+            let queryString = currentUrl ? currentUrl.split('?')[1] : window.location.search.slice(1)
+            let url = `/sass/download-sass-taxon-data/?csvName=${csv_name}&${queryString}`;
+            showDownloadPopup('CSV', csv_name, function (downloadRequestId) {
+                url += `&downloadRequestId=${downloadRequestId}`;
+                downloadButton.html("Processing...");
+                downloadButton.prop("disabled", true);
+                const alertModalBody = $('#alertModalBody');
+                if (!is_logged_in) {
+                    alertModalBody.html('Please log in first.')
+                }
```

Comment:
Please add this message if user is logged in after clicking download : 
'Your data download is underway. ' +
          'This may take some time. ' +
          'You will be notified by email when your download is ready. ' +
          'Thank you for your patience.'